operator/controllers/csiscaleoperator_controller.go

```diff
@@ -219,6 +221,62 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	logger.Info("Creation of the resources which never change is successful")
 
+	// Rollout restart of node plugin pods, if modified driver
+	// manifest file is applied.
+	// The CSIConfigMap has the data which is last applied, so
+	// compare the clusters data of current driver instance with
+	// CSIConfigMap data and rollout restart if there is a difference.
+	configMap := &corev1.ConfigMap{}
+	cerr := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      config.CSIConfigMap,
+		Namespace: req.Namespace,
+	}, configMap)
+	if cerr != nil {
+		if !errors.IsNotFound(cerr) {
+			message := "Failed to get ConfigMap resource " + config.CSIConfigMap
+			logger.Error(err, message)
+			// TODO: Add event.
+			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+				Type:    string(config.StatusConditionSuccess),
+				Status:  metav1.ConditionFalse,
+				Reason:  string(csiv1.ResourceReadError),
+				Message: message,
+			})
+			logger.Error(err, message)
+			return ctrl.Result{}, cerr
+		}
+	} else {
+		clustersBytes, err := json.Marshal(&instance.Spec.Clusters)
+		if err != nil {
+			logger.Error(err, "Failed to marshal clusters data of this instance")
+			return ctrl.Result{}, err
+		}
+		clustersString := string(clustersBytes)
+		configMapDataBytes, err := json.Marshal(&configMap.Data)
+		if err != nil {
+			logger.Error(err, "Failed to marshal data of ConfigMap "+config.CSIConfigMap)
+			return ctrl.Result{}, err
+		}
+		configMapDataString := string(configMapDataBytes)
+		configMapDataString = strings.Replace(configMapDataString, " ", "", -1)
+		configMapDataString = strings.Replace(configMapDataString, "\\\"", "\"", -1)
+
+		if !strings.Contains(configMapDataString, clustersString) {
+			logger.Info("CSIScaleOperator instance is changed, so restarting node plugin pods")
```

Comment:
Since we are triggering restart here when cluster field is modified and not other fields, we can include it in the log as well.

Reply:
Done